wordpress-mobile/WordPress-Android
# Fixes Blurred Featured Image on Post Settings for Self-Hosted sites

Fixes #15926

## Description
On self-hosted sites [the `getResizedImageUrl`](https://github.com/wordpress-mobile/WordPress-Android/pull/16439/files#diff-8dbc630ce5f70af93b292d1a57082544599c3800174de6c017bb1b74c48e9d4cR171) does not have any actual effect on the resulting image size. Thus the default low resolution thumbnail image is used (default 150x150) causing the reported issue. E.g. the https://overwhelming-falcon.jurassic.ninja/wp-content/uploads/2022/04/PXL_20220427_072207344-2-150x150.jpg?w=413&h=413 does not provide a resized image.
To solve this I bypass this mechanism only for self-hosted sites and always use the full-size media url.

## To test:
1. On a self-hosted/Atomic site create a new post (or open an existing one to edit it).
2. Tap Menu > Post settings
3. If the post has a featured image **verify** that it is not blurred
4. If the post didn't have a Featured Image yet, tap "Set Featured Image"
5. Choose any images, either uploading a new one or choose from Media Library
6. **Verify** that the image is not blurred

## Regression Notes
1. Potential unintended areas of impact
Regression for .com sites

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit tests

3. What automated tests I added (or what prevented me from doing so)
Relied on existing unit test and added a new test case in `FeaturedImageHelperTest`

PR submission checklist:
